soh/soh/Enhancements/randomizer/hook_handlers.cpp

```diff
@@ -1483,6 +1574,284 @@ void RandomizerOnActorInitHandler(void* actorRef) {
+    }
+
+    if (!Flags_GetRandomizerInf(RAND_INF_HAS_WALLET)) {
+        gSaveContext.rupees = 0;
+    }
+}
+
+extern "C" void func_8099485C(DoorGerudo* gerudoDoor, PlayState* play);
+
+void RandomizerOnActorUpdateHandler(void* refActor) {
+    Actor* actor = static_cast<Actor*>(refActor);
+
+    if (Flags_GetRandomizerInf(RAND_INF_HAS_SKELETON_KEY)) {
+        if (actor->id == ACTOR_EN_DOOR) {
+            EnDoor* door = reinterpret_cast<EnDoor*>(actor);
+            door->lockTimer = 0;
+        } else if (actor->id == ACTOR_DOOR_SHUTTER) {
+            DoorShutter* shutterDoor = reinterpret_cast<DoorShutter*>(actor);
+            if (shutterDoor->doorType == SHUTTER_KEY_LOCKED) {
+                shutterDoor->unk_16E = 0;
+            }
+        } else if (actor->id == ACTOR_DOOR_GERUDO) {
+            DoorGerudo* gerudoDoor = (DoorGerudo*)actor;
+            gerudoDoor->actionFunc = func_8099485C;
+            gerudoDoor->dyna.actor.world.pos.y = gerudoDoor->dyna.actor.home.pos.y + 200.0f;
+        }
+    }
+
+    // In ER, override the warp song locations. Also removes the warp song cutscene
+    if (RAND_GET_OPTION(RSK_SHUFFLE_ENTRANCES) && actor->id == ACTOR_DEMO_KANKYO && actor->params == 0x000F) { // Warp Song particles
+        Entrance_SetWarpSongEntrance();
+    }
+
+    if (actor->id == ACTOR_OBJ_COMB) {
+        ObjComb* combActor = reinterpret_cast<ObjComb*>(actor);
+        combActor->actor.shape.rot.x = Math_SinS(combActor->unk_1B2) * CLAMP_MIN(combActor->unk_1B0, 0) + combActor->actor.home.rot.x;
+    }
+}
+
+//from z_player.c
+typedef struct {
+    /* 0x00 */ Vec3f pos;
+    /* 0x0C */ s16 yaw;
+} SpecialRespawnInfo; // size = 0x10
+
+//special respawns used when voided out without swim to prevent infinite loops
+std::map<s32, SpecialRespawnInfo> swimSpecialRespawnInfo = {
+    {
+        ENTR_ZORAS_RIVER_3,//hf to zr in water
+        { { -1455.443, -20, 1384.826 }, 28761 }
+    },
+    {
+        ENTR_HYRULE_FIELD_14,//zr to hf in water
+        { { 5830.209, -92.16, 3925.911 }, -20025 }
+    },
+    {
+        ENTR_LOST_WOODS_7,//zr to lw
+        { { 1978.718, -36.908, -855 }, -16384 }
+    },
+    {
+        ENTR_ZORAS_RIVER_4,//lw to zr
+        { { 4082.366, 860.442, -1018.949 }, -32768 }
+    },
+    {
+        ENTR_LAKE_HYLIA_1,//gv to lh
+        { { -3276.416, -1033, 2908.421 }, 11228 }
+    },
+    {
+        ENTR_WATER_TEMPLE_0,//lh to water temple
+        { { -182, 780, 759.5 }, -32768 }
+    },
+    {
+        ENTR_LAKE_HYLIA_2,//water temple to lh
+        { { -955.028, -1306.9, 6768.954 }, -32768 }
+    },
+    {
+        ENTR_ZORAS_DOMAIN_4,//lh to zd
+        { { -109.86, 11.396, -9.933 }, -29131 }
+    },
+    {
+        ENTR_LAKE_HYLIA_7,//zd to lh
+        { { -912, -1326.967, 3391 }, 0 }
+    },
+    {
+        ENTR_GERUDO_VALLEY_1,//caught by gerudos as child
+        { { -424, -2051, -74 }, 16384 }
+    }
+};
+
+f32 triforcePieceScale;
+
+void RandomizerOnPlayerUpdateHandler() {
```

Comment:
We should likely split this into two separate hook handlers, both of them only being registered when their respective options are on (triforce hunt & the randomized swimming)

Reply:
I was thinking that it might be better to have only one hook handler for each hook in randomizer and split if off inside the hook to prevent registering too many hooks, if it's preferred to use separate hooks, I'll change it to that.